packages/babel-core/src/config/files/types.js

```diff
@@ -3,7 +3,7 @@
 export type ConfigFile = {
   filepath: string,
   dirname: string,
-  options: {},
+  options: Object,
```

Comment:
Still playing around with improving this... but everything so far is quite nasty (json5 returning mixed =/)

Reply:
Does this just start failing with the new Flow version then? I was definitely under the impression everything should refine fully to `{}` with the existing logic.

Reply:
Yeah new from the version bump

Reply:
Hmm, let me look quickly, I'd rather keep these types accurate.

Reply:
The failures is
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ packages/babel-core/src/config/files/configuration.js:202:16

Property babel is missing in object type [1].

     199│ const packageToBabelConfig = makeWeakCache(
     200│   (file: ConfigFile): ConfigFile | null => {
 [1] 201│     const options: {} = file.options;
     202│     if (typeof options.babel === "undefined") return null;
     203│     const babel = options.babel;
     204│
     205│     if (typeof babel !== "object" || Array.isArray(babel) || babel === null) {



Found 1 error
```
it looks like we can change that code to
```
const babel = file.options[("babel": string)];
if (typeof babel === "undefined") return null;
```
to get it to typecheck safely.